### PR TITLE
Task 13

### PR DESCRIPTION
**Pull request by 233gme**

### CODE REVIEW

File: src/Components/CardList/Card/index.jsx [hunk 1]
```diff
@@ -1,71 +1,55 @@
...
-      ...props.card,
-      editModeFlag: false
-    })
-  }
+    props.onSaveChanges({...props.card, editModeFlag: false});
+    setNewState({...props.card, editModeFlag: false})
+  };
+
+  const toFullCardPage = () => {
+    if (!newState.editModeFlag && !props.onView) {
```
Comment (iba-dhetsikau) [inline src/Components/CardList/Card/index.jsx]: Why don't you allow to open Card page in View Mode? Page should be accessible, simply in a view mode only.  Think about moving View Mode state to redux as well.

File: src/Components/CardList/Card/CardHeader/index.jsx [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
   const defaultMode = (
     <div className='card_title'>
       <p>{props.card.title}</p>
-      <div>
-        {!props.onView && <button title='режим редактирования' onClick={props.editModeOn}><AiFillEdit /></button>}
-        <input className='card_checbox' type='checkbox' onClick={props.selectItem} />
+      <div className='card_title__btn_block'>
+        <Route path='/' exact>
+          <input className='card_checbox' type='checkbox' onClick={props.selectItem} />
+        </Route>
```
Comment (iba-dhetsikau) [inline src/Components/CardList/Card/CardHeader/index.jsx]: Is there any specific reason to add this route? I do not see any. You can pass a property saying of it's a card is viewed in a list or on a page and check it in order to render the checkbox.

File: src/Components/CardList/index.jsx [hunk 3]
```diff
@@ -1,21 +1,28 @@
...
-            card={card}
-            onView={view}
-            onSaveChanges={onSave} />
-        )
-      })}
+          key={card.id}
+          card={card}
+          onView={view}
+          onSaveChanges={(val) => onSaveChanges(val)}
+          route={props} />
```
Comment (iba-dhetsikau) [inline src/Components/CardList/index.jsx]: Passing all the component props into a nested component - bad idea, there could be many redundant ones. Pass only the ones that are really required.

File: src/Components/Pages/FullCard/index.jsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    else props.history.push('/404')
+  }, [cards, props.match.params.id, props.history])
+
+  return (
+    <div className='fullcard_block'>
+      <Card
+        card={fullcard}
+        onView={view}
+        onSaveChanges={(val) => onSaveChanges(val)}
+        route={props} />
```
Comment (iba-dhetsikau) [inline src/Components/Pages/FullCard/index.jsx]: The same as before. Better pass only the required props.

File: src/Components/Pages/SingIn/index.jsx [hunk 5]
```diff
@@ -20,7 +19,7 @@ const SignIn = () => {
     event.preventDefault();
     setFormState({ username: '', password: '', });
     setSubmit(false);
-    console.log(`Form Submit`)
+    console.log('[SingIn] - ',formState)
```
Comment (iba-dhetsikau) [inline src/Components/Pages/SingIn/index.jsx]: Actually all debug info should be excluded from the code. unless it described some error or exceptional case.

**Review by iba-dhetsikau — CHANGES_REQUESTED** [review]
Please apply changes according to file comments and resend for a review.

File: src/Components/CardList/Card/index.jsx (hunk 1 above)
Comment (233gme) [reply src/Components/CardList/Card/index.jsx]: Fixed

File: src/Components/CardList/Card/CardHeader/index.jsx (hunk 2 above)
Comment (233gme) [reply src/Components/CardList/Card/CardHeader/index.jsx]: I decided that this function does not make sense on the "card /id" page.
Fixed

File: src/Components/CardList/index.jsx (hunk 3 above)
Comment (233gme) [reply src/Components/CardList/index.jsx]: Fixed

File: src/Components/Pages/FullCard/index.jsx (hunk 4 above)
Comment (233gme) [reply src/Components/Pages/FullCard/index.jsx]: Fixed

File: src/Components/Pages/SingIn/index.jsx (hunk 5 above)
Comment (233gme) [reply src/Components/Pages/SingIn/index.jsx]: Ok, deleted

**Review by iba-dhetsikau — APPROVED** [review]
Merge approved.

**Pull request merged**